README.md

```diff
@@ -1,8 +1,161 @@
+
+## Installation
+
+At the Julia prompt, add the `GigaSOM` package:
+
+```julia
+julia> import Pkg; Pkg.add("GigaSOM");
+```
+
+Use the `GigaSOM` module by running:
+
+```julia
+julia> using GigaSOM
+```
+
+## Test the installation
+
+`GigaSOM` has been tested on Linux (Ubuntu and centOS), macOS, and Windows.
+
+You may test the package as follows:
+
+```julia
+julia> import Pkg; Pkg.test("GigaSOM");
+```
+
+Alternatively, you can use the package manager by hitting `]`:
+
+```julia
+(v1.1) pkg> test GigaSOM
+```
+
+:warning: It is not advised to run indivual test files separately without expliciting activating the environment and loading the dependencies.
+If this is required for debugging purposes, please activate the environment first and run the commands of the `test/runtests.jl` file sequentially.
+
+
+# How to contribute to/develop GigaSOM
+
+If you want to contribute to the `GigaSOM` package, please fork the present repository. Instructions how to this can be found [here](https://help.github.com/en/articles/fork-a-repo).
+
+## Step 1: Retrieve a local version of GigaSOM
+
+There are 2 ways that you can retrieve a local copy of the package: one is to manually clone the forked repository, and the second one is to use the intergrated Julia package manager.
+
+### Option 1: Manually clone the fork
+
+You can do this as follows from the command line:
+
+```bash
+$ git clone git@github.com:yourUsername/GigaSOM.jl.git GigaSOM.jl
+$ cd GigaSOM.jl
+$ git checkout -b yourNewBranch origin/develop
+```
+
+where `yourUsername` is your Github username and `yourNewBranch` is the name of a new branch.
+
+Then, in order to develop the package, you can install your cloned version as follows (make sure you are in the `GigaSOM.jl` directory):
+
+```julia
+(v1.1) pkg> add .
+```
+
+This adds the `GigaSOM.jl` package and all its dependencies. You can verify that the installation worked by typing:
+
+```julia
+(v1.1) pkg> status
+```
+
+If everything went smoothly, this should print something similar to:
+
+```julia
+(v1.1) pkg> status
+    Status `~/.julia/environments/v1.1/Project.toml`
+  [a03a9c34] GigaSOM v0.0.5 #yourNewBranch (.)
+```
+
+Now, you can readily start using the `GigaSOM` module:
+
+```julia
+julia> using GigaSOM
+```
+
+### Option 2: Use the Julia package manager
+
+When you are used to using the  Julia package manager for developing or contributing to packages, you can type:
+
+```julia
+(v1.1) pkg> dev GigaSOM
+```
+
+This will install the `GigaSOM` package locally and check it out for development. You can check the location of the package with:
+
+```julia
+(v1.1) pkg> status
+    Status `~/.julia/environments/v1.1/Project.toml`
+  [a03a9c34] GigaSOM v0.0.5 [`~/.julia/dev/GigaSOM`]
+```
+
+The default location of the package is `~/.julia/dev/GigaSOM`.
+
+You can then set your remote by:
+
+```bash
```

Comment:
Maybe mention explicitly that these commands should be executed in a regular shell.

Reply:
And that the repo `yourUsername/GigaSOM.jl` needs to exist already.